### ISSUE TITLE
[QUESTION] Problem with images from S3 

### ISSUE THREAD

**Issue by gagaha** [RocketChat/Rocket.Chat.iOS#2129]
- Rocket.Chat app version:  3.0.4 (206)
- Rocket.Chat server version: 0.68.4
- Device: iPhone 7 Plus (iOS 11.4.1)

The problem occurs when using S3 (in my case Ceph RGW) as file upload storage.

"Proxy Uploads" is disabled in the settings on the RC server.
Upload a video and it will be playable from the app.
Upload a Image and it will not be displayed or viewable.
 
Enable "Proxy Uploads" and it is exactly the other way around:
Videos do not play but images will be displayed.

**Comment by serge-petrov** [comment]
I have the same issue.
Server version is 0.68.3
Also some text messages are delayed when using iOS application.
Please note that web application works fine.

**Comment by rafaelks** [comment]
@gagaha @serge-petrov Can you guys post an example of link to an image, or give us access to the server?

**Comment by gagaha** [comment]
@rafaelks Thanks for looking into this! Here is a example link to an [image](https://rgw1.netways.de/1826-rocketchat-ffe65/1826-rocketchat-ffe65/anE4DKhoL4cMvjEbG/uploads/XSgP6M9aZjGouCruX/5XYKsEreG4kMwZmeE/DfymPwsMyJwiPnXSJ?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ZXPJ4840TKBFPBY5D8PY%2F20180904%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180904T073328Z&X-Amz-Expires=120&X-Amz-Signature=173b01065bc9fe5a1f6e325540539619645814736146711c46970cf5287a7713&X-Amz-SignedHeaders=host)

I'll send you an e-mail to give you access to my test-server.

**Comment by rafaelks** [comment]
@gcommit I still didn't have time to look into the issue and investigate, we'll probably be able to do that later next week.

**Comment by gcommit** [comment]
Hey @rafaelks any news on this issue ?

**Comment by rafaelks** [comment]
@gcommit I'm sorry the delay here... did you send me the email with your credentials? Can you send it again please?

**Comment by rafaelks** [comment]
This issue was solved already. 👍

**Comment by rafaelks** [comment]
@gagaha It was a bad Nginx configuration, but maybe @gcommit can give you the proper solution, since he was the one that fixed it on his side.

**Comment by gcommit** [comment]
@rafaelks, ehm, i did not resolve that o.O

**Comment by rafaelks** [comment]
I guess was @serge-petrov then? 🤔

**Comment by gagaha** [comment]
@serge-petrov please let us know how you solved it.

**Comment by rafaelks** [comment]
@Fabster79 Can you describe better your problem? You can't see S3 image uploads on the app?

**Comment by Fabster79** [comment]
Yeah, the images are blank on my iphone, but i can see them on web. 

Iphone app:
![image](https://user-images.githubusercontent.com/29672511/51117765-fde89e00-180e-11e9-9c68-492b61303cfb.png)



web
![image](https://user-images.githubusercontent.com/29672511/51117686-caa60f00-180e-11e9-863f-723d0390903f.png)

**Comment by gagaha** [comment]
I noticed that it is possible to view the images if you touch the name of the chat-room and go to "Files".
"Proxy uploads" have to be disabled in the S3 settings of the Rocket.Chat server in order to view the images there.
But you will still not be be able to view the images in the chat history.

@rafaelks could you have a look at this for us? Thanks!

**Comment by rafaelks** [comment]
I'm going to investigate this issue, thanks for the input guys! 👍

**Comment by rafaelks** [comment]
@gagaha @Fabster79 Could you guys confirm to me if the setting `CDN_PREFIX` has anything for you guys?

**Comment by rafaelks** [comment]
@gagaha @Fabster79 We're getting close to another release, and your feedback here would be essential to get this fixed for version 3.3. Thanks!

**Comment by gagaha** [comment]
@rafaelks Sorry for the delay! Settings for CDN Prefix is empty in my Rocket.Chat.

**Comment by rafaelks** [comment]
@gagaha Could you talk to me in `open.rocket.chat`? My username is `rafael.kellermann`.

**Comment by gagaha** [comment]
This is has been resolved.

To sum up the solution:
The webserver (Civetweb) of the S3 storage (Ceph Rados Gateway) allowed some weak ssl cipher suites.
Since Apple is enforcing PFS for all SSL connections of iOS Apps, this was the root of the problem.
I could not get PFS working with Civetweb until I did a dist-upgrade of Ubuntu (14.04. to 16.04).
Using a Proxy (Apache/Nginx) would have been also a solution.
This is the cipher list I am using now:
```
EECDH+ECDSA+AESGCM:EECDH+aRSA+AESGCM:EECDH+ECDSA+SHA384:EECDH+ECDSA+SHA256:EECDH+aRSA+SHA384:EECDH+aRSA+SHA256:EECDH+aRSA+RC4:EECDH:EDH+aRSA:RC4:!aNULL:!eNULL:!LOW:!3DES:!MD5:!EXP:!PSK:!SRP:!DSS:!MEDIUM
```
I'm getting an A-Rating for SSL now and images and videos are displayed as expected.

Thanks @rafaelks for your help!
